src/WingetCreateCLI/PromptHelper.cs

```diff
@@ -0,0 +1,270 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.WingetCreateCLI
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.ComponentModel.DataAnnotations;
+    using System.Linq;
+    using System.Reflection;
+    using Microsoft.WingetCreateCLI.Logging;
+    using Microsoft.WingetCreateCLI.Properties;
+    using Microsoft.WingetCreateCore;
+    using Microsoft.WingetCreateCore.Common;
+    using Microsoft.WingetCreateCore.Models.Installer;
+    using Newtonsoft.Json;
+    using Sharprompt;
+
+    /// <summary>
+    /// Provides functionality for prompting the user for input to obtain field values for a manifest.
+    /// </summary>
+    public static class PromptHelper
+    {
+        /// <summary>
+        /// List of strings representing the optional fields that should not be editable.
+        /// </summary>
+        private static readonly string[] NonEditableOptionalFields = new[]
+        {
+            nameof(InstallerType),
+            nameof(InstallerManifest.Channel),
+            nameof(Installer.InstallerUrl),
+            nameof(Installer.InstallerSha256),
+            nameof(Installer.SignatureSha256),
+            nameof(Installer.ProductCode),
+            nameof(Installer.PackageFamilyName),
+            nameof(Installer.AdditionalProperties),
+        };
+
+        /// <summary>
+        /// Displays all properties of a model as a navigational menu selection with some additional filtering logic based on InstallerType.
+        /// </summary>
+        /// <typeparam name="T">Model type.</typeparam>
+        /// <param name="model">Instance of object model.</param>
+        /// <param name="exitMenuWord">Exit keyword to be shown to the user to exit the navigational menu.</param>
+        public static void DisplayPropertiesAsMenuSelection<T>(T model, string exitMenuWord)
+        {
+            Console.Clear();
+
+            var properties = model.GetType().GetProperties().ToList();
+            var optionalProperties = properties.Where(p =>
+                p.GetCustomAttribute<RequiredAttribute>() == null &&
+                p.GetCustomAttribute<JsonPropertyAttribute>() != null).ToList();
+
+            var fieldList = properties
+                .Select(property => property.Name)
+                .Where(pName => !NonEditableOptionalFields.Any(field => field == pName))
+                .Append(exitMenuWord)
+                .ToList();
+
+            var installerTypeProperty = model.GetType().GetProperty(nameof(InstallerType));
+            if (installerTypeProperty != null)
+            {
+                var installerType = installerTypeProperty.GetValue(model);
+                if (installerType != null)
```

Comment:
There was more than just these two, see full comment in spec:

James would know better than me, but I'm not sure if Scope applies to msix? Can you have a machine-scope msix?

Can Platform apply to anything but MSIX (isn't everything else just "Desktop"?) PackageFamilyName is MSIX-only.

A few things don't apply to MSIX since they don't have install command lines. Things like Switches, success codes, maybe also UpgradeBehavior? InstallModes?

ProductCode is MSI-only (or "exe", since it may have an embedded MSI)

Reply:
Added an inclusion and exclusion list to handle this special cases. I haven't messed with Platform as I need further clarification. Do you mean that we should be excluding Platform for everything but MSIX and setting Platform: Desktop by default for all other installers except MSIX? Or is Platform:Desktop the default for all other installers regardless of whether its present in the manifest.

Reply:
I think we need clarification from @denelon on Platform. I'm not sure when it would be anything but Desktop. Also not sure if it should be editable. For MSIX we get it from the appxmanifest, and I'm not sure if it applies to any other installer types.

PackageFamilyName shouldn't be editable, we get it from the appxmanifest.

ProductCode should be editable for "exe" installer type too, due to embedded MSIs.

Reply:
Platform is in place for a future where we might be installing on other devices like HoloLens, XBox or the next computing platform. I think it would be pretty awesome to be able to `winget install <package>` on a quantum computer 😊.

Reply:
I removed PackageFamilyName from msix since it shouldn't be editable and we can get that from appxmanifest.

Since platform isn't fully ready yet, I made it hidden for now for all installers except MSIX.

I also allowed ProductCode to be editable if installertype is an EXE.

Reply:
Cool. So @ryfu-msft I think you made the right choice to just make Platform non-editable for now. Someone can set it in their appx manifest, but we shouldn't allow it to be editable

Reply:
Got it, so it looks like the short answer is don't make platform editable at all for anything right now. Removed platform from editable list.